src/accessories/CampaignImporterExporter.ttslua

```diff
@@ -0,0 +1,292 @@
+    position = {x=-1, y=0.2, z=0},
+    font_size = 400,
+    width = 1400,
+    height = 600,
+    scale = {0.5, 1, 0.5},
+  })
+  self.createButton({
+    click_function = "createCampaignToken",
+    function_owner = self,
+    label = "Export",
+    tooltip = "Create a campaign save token!\n\n(Ensure all chaos tokens have been unsealed!)",
+    position = {x=1, y=0.2, z=0},
+    font_size = 400,
+    width = 1400,
+    height = 600,
+    scale = {0.5, 1, 0.5},
+  })
+end
+
+-- The main import functions.  Due to timing concerns, has to be split up into several separate methods to allow for Wait conditions
+
+-- Identifies import token, determines campaign box and downloads it (if needed)
+function findCampaignFromToken(_, _, _)
+  local coin = nil
+  local coinObjects = getObjectsWithTag("ImporterToken")
+  if #coinObjects == 0 then
+    broadcastToAll("Could not find importer token", Color.Red)
+  elseif #coinObjects > 1 then
+    broadcastToAll("More than 1 importer token found. Please delete all but 1 importer token", Color.Yellow)    
+  else
+    coin = coinObjects[1]
+    local importData = JSON.decode(coin.getGMNotes())
+    campaignBoxGUID = importData["box"]
+    local campaignBox = getObjectFromGUID(campaignBoxGUID)
+    if campaignBox.type == "Generic" then
+      campaignBox.call("buttonClick_download")
+    end
+    Wait.condition(
+      function()
+        if #campaignBox.getObjects() > 0 then
+          placeCampaignFromToken(importData)
+        else
+          createCampaignFromToken(importData)
+        end
+      end,
+      function()
+        local obj = getObjectFromGUID(campaignBoxGUID)
+        if obj == nil then 
+          return false 
+        else
+          return obj.type == "Bag" and obj.getLuaScript() ~= ""
+        end
+      end,
+      2,
+      function() broadcastToAll("Error loading campaign box") end
+    ) 
+  end
+end
+
+-- After box has been downloaded, places content on table
+function placeCampaignFromToken(importData)
+  getObjectFromGUID(campaignBoxGUID).call("buttonClick_place")
+  Wait.condition(
+    function() createCampaignFromToken(importData) end,
+    function() return findCampaignLog() ~= nil end,
+    2,
+    function() broadcastToAll("Error placing campaign box") end
+  )
+end
+
+-- After content is placed on table, conducts all the other import operations
+function createCampaignFromToken(importData)
+  findCampaignLog().destruct()
+  --create campaign log
+  spawnObjectData({data = importData["log"]}) 
+  --create chaos bag
+  chaosBagApi.setChaosBagState(importData["bag"])
+  --populate trauma values
+  if importData["trauma"] then
+    updateCounters(importData["trauma"])
+  end
+  --populate ArkhamDB deck IDs
+  if importData["decks"] then
+    deckImporterApi.setUiState(importData["decks"])
+  end
+  --set investigator count
+  playAreaApi.setInvestigatorCount(importData["clueCount"])
+  --set campaign guide page
+  if findCampaignGuide() then
```

Comment:
I'd save the return value from `findCampaignGuide` to a variable, so you don't have to call it multiple times